examples/methods/plot_reweighting.py

```diff
@@ -0,0 +1,403 @@
+    DecisionBoundaryDisplay.from_estimator(
+        clf,
+        Xs,
+        cmap=ListedColormap(["w", "k"]),
+        alpha=1,
+        ax=ax,
+        eps=0.5,
+        response_method="predict",
+        plot_method="contour",
+    )
+
+    ax.set_xticks(()), ax.set_yticks(())
+    ax.set_xlim(x_min, x_max), ax.set_ylim(y_min, y_max)
+    ax.set_title("Training with rewegihted data", fontsize=12)
+    figure.suptitle(suptitle, fontsize=16, y=1)
+
+
+clf = base_classifier
+clf.fit(Xs, ys)
+create_plots(
+    base_classifier,
+    name="Without DA",
+    weights=np.array([2] * Xs.shape[0]),
+    suptitle="Illustration of the classifier with no DA",
+)
+
+# %%
+#     Illustration of the Reweight Density method
+# ------------------------------------------
+#
+# This method is trying to compute the optimal weights as a ratio of two probability
+# functions, by default, it is the ratio of two kernel densities estimations.
+
+# We define our classifier, `clf` is a da pipeline
+clf = ReweightDensity(
+    base_estimator=base_classifier,
+    weight_estimator=KernelDensity(bandwidth=0.5),
+)
+clf.fit(X, y, sample_domain=sample_domain)
+
+# We get the weights:
+
+# we first get the adapter which is estimating the weights
+weight_estimator = clf[0].base_estimator
+weight_estimator.fit(X, sample_domain=sample_domain)
+idx = extract_source_indices(sample_domain)
+weights = weight_estimator.adapt(X, sample_domain=sample_domain).sample_weight[idx]
+
+create_plots(clf, weights=weights, name="Reweight Density")
+
+# %%
+#     Illustration of the Gaussian reweighting method
+# ------------------------------------------
+# This method tries to approximate the optimal weights by assuming that the data are
+# normally distributed, and thus approximating the probability functions for both source
+# and target set, and setting the weight to be the ratio of the two.
+#
+# See [2] for details.
+#
+# .. [2]  Hidetoshi Shimodaira. Improving predictive inference under
+#         covariate shift by weighting the log-likelihood function.
+#         In Journal of Statistical Planning and Inference, 2000.
+
+# We define our classifier, `clf` is a da pipeline
+clf = GaussianReweightDensity(base_classifier)
+clf.fit(X, y, sample_domain=sample_domain)
+# We get the weights
+weight_estimator = clf[0].base_estimator
+weight_estimator.fit(X, sample_domain=sample_domain)
+idx = extract_source_indices(sample_domain)
+weights = weight_estimator.adapt(X, sample_domain=sample_domain).sample_weight[idx]
+
+create_plots(clf, weights=weights, name="Gaussian Reweight Density")
+
+# %%
+#     Illustration of the Discr. reweighting method
+# ------------------------------------------
+#
+# This estimator derive a class of predictive densities by weighting the source samples
+# when trying to maximize the log-likelihood function. Such approach is effective in
+# cases of covariate shift.
+#
+# See [3] for details.
+#
+# .. [3]    Hidetoshi Shimodaira. Improving predictive inference under
```

Comment:
Is it the same as [2]

Reply:
yes, in `_reweight.py` it is the same also